consensus/core/src/core.rs

```diff
@@ -935,11 +935,64 @@ impl Core {
             }
         }
 
-        assert!(parent_round_quorum.reached_threshold(&self.context.committee), "Fatal error, quorum not reached for parent round when proposing for round {clock_round}. Possible mismatch between DagState and Core.");
-
+        // Inclusion of weak links for low score ancestors so there is no long
+        // list of blocks that we need to include later.
         for (score, ancestor) in excluded_ancestors.iter() {
             let excluded_author = ancestor.author();
             let block_hostname = &self.context.committee.authority(excluded_author).hostname;
+            // Use the low quorum accepted round as reported by the network for which
+            // round to include for this ancestor.
+            let mut network_low_quorum_round = self
+                .ancestor_state_manager
+                .accepted_quorum_round_per_authority[excluded_author]
+                .0;
+
+            // If the network quourum round for this ancestor is greater than or equal
+            // to the clock round then we want to make sure to set it to clock_round - 1
+            // as that is the max round we can include as an ancestor.
+            network_low_quorum_round = network_low_quorum_round.min(quorum_round);
+
+            if let Some(last_block_ref) = self.last_included_ancestors[excluded_author] {
+                if last_block_ref.round < network_low_quorum_round {
+                    if ancestor.round() == network_low_quorum_round {
```

Comment:
1. We should merge this if code block with below to avoid duplicated logic.
2. Instead of using `get_uncommitted_blocks_at_slot()` below, I think we need to implement another version of `get_last_block_for_authority()` with an additional end round arg.

Reply:
> 2. Instead of using get_uncommitted_blocks_at_slot() below, I think we need to implement another version of get_last_block_for_authority() with an additional end round arg.

I do have the exact same comment and actually that's a comment I did on the original PR https://github.com/MystenLabs/sui/pull/19605#discussion_r1789978267 - I believe that @arun-koshy  did already implement this back then before removing the weak link inclusion

Reply:
Oh right this commit should have all of the relevant changes before I remove weak link inclusion. I will wait for Mingwei's changes first and then see what is left to be cleaned up

https://github.com/MystenLabs/sui/commit/cbf078910e6b768e80e461b45247c40ae72c3d55#diff-c3e7a1c4d7f602a38d2146e54671ab744e30e91e3b79b28fba5f598536a49b68